Kernel/Arch/riscv64/Interrupts.cpp

```diff
@@ -76,7 +76,7 @@ extern "C" void trap_handler(TrapFrame& trap_frame)
                     auto* handler = s_interrupt_handlers[pending_interrupt];
                     VERIFY(handler);
                     handler->increment_call_count();
-                    handler->handle_interrupt(*trap_frame.regs);
```

Comment:
Line 71 also needs the RegisterState argument removed (we should really get RISC-V CI). Other than that, I do quite like this change.